radicle/src/cob/patch.rs

```diff
@@ -885,6 +885,27 @@ impl<'a> Patches<'a> {
         Ok(state_groups)
     }
 
+    /// Find the proposed patch containing the given `Revision`.
+    pub fn find_proposed_by_revision(
+        &self,
+        id: &RevisionId,
+    ) -> Result<Option<(PatchId, Patch)>, Error> {
```

Comment:
This function could be improved by taking a `state` parameter, it's currently overly specific for an API function.